### PR TITLE
Unary Ops Grad: Adds gradient for dl.step

### PR DESCRIPTION
**Pull request by ManrajGrover**
This PR adds gradient for `dl.step`

---
This change is [Reviewable](https://reviewable.io/reviews/pair-code/deeplearnjs/672)

### CODE REVIEW

**Comment by dsmilkov** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[src/math/unary_ops.ts, line 48 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/672#-L4ll-YgNzhst7cfBmvO:-L4ll-YgNzhst7cfBmvP:b-p7xd54) ([raw file](https://github.com/pair-code/deeplearnjs/blob/05b52a4d6dec087c36539ae76f780c13de2c33df/src/math/unary_ops.ts#L48)):*
> ```XML
>   @operation
>   static ceil<T extends Tensor>(x: T): T {
>     // TODO(manrajgrover): Fix gradients for the case where x == ceil(x)
> ```

there is no need to fix gradients for x == ceil(x). We are following tensorflow's decision to have grad(ceil) always be zeros, regardless of `x`. You can change the TODO(manrajgrover): Return null for gradients when backprop supports it.

---

*[src/math/unary_ops.ts, line 62 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/672#-L4llLDZaOKV2ikxvc33:-L4llLDZaOKV2ikxvc34:b-njtz48) ([raw file](https://github.com/pair-code/deeplearnjs/blob/05b52a4d6dec087c36539ae76f780c13de2c33df/src/math/unary_ops.ts#L62)):*
> ```XML
>   @operation
>   static floor<T extends Tensor>(x: T): T {
>     // TODO(manrajgrover): Fix gradients for the case where x == floor(x)
> ```

same here. see comment above

---

*[src/math/unary_ops.ts, line 390 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/672#-L4llPPzEm3JPLbZ6U57:-L4llPPzEm3JPLbZ6U58:b-8jd94f) ([raw file](https://github.com/pair-code/deeplearnjs/blob/05b52a4d6dec087c36539ae76f780c13de2c33df/src/math/unary_ops.ts#L390)):*
> ```XML
>   @operation
>   static step<T extends Tensor>(x: T, alpha = 0.0): T {
>     // TODO(manrajgrover): Fix gradients for the case where x == 0
> ```

same here. TODO should be for returning `null` later.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/672)*

**Comment by dsmilkov** [comment]
:lgtm_strong: Great!

---

Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/672#-:-L5ACisICf_-C_XxrVWx:b-a352bd)*
